subsys/bluetooth/audio/ascs.c

```diff
@@ -985,61 +1063,47 @@ void ascs_ep_init(struct bt_audio_ep *ep, uint8_t id)
 
 static void ase_init(struct bt_ascs_ase *ase, uint8_t id)
 {
-	memset(ase, 0, sizeof(*ase));
 	ascs_ep_init(&ase->ep, id);
 
 	/* Lookup ASE characteristic */
 	bt_gatt_foreach_attr_type(0x0001, 0xffff, ASE_UUID(id), NULL, 0, ase_attr_cb, ase);
 
-	__ASSERT(ase->ep.server.attr, "ASE characteristic not found\n");
+	__ASSERT(ase->attr, "ASE characteristic not found\n");
 }
 
 static struct bt_ascs_ase *ase_new(struct bt_ascs *ascs, uint8_t id)
 {
 	struct bt_ascs_ase *ase;
-	int i;
 
-	if (id) {
-		if (id > ASE_COUNT) {
-			return NULL;
-		}
-		i = id;
-		ase = &ascs->ases[i - 1];
-		goto done;
+	if (!id || id > ASE_COUNT) {
+		return NULL;
```

Comment:
So the change here is that `ase_new` should never be called with `id == 0`, right? Previously implementation used `id == 0` to get a new ASE, but that's no longer the case. 

If we never want to call this with `id == 0`, I suggest to change this check to an __ASSERT instead, as returning NULL would probably just make it more confusing to use.

Reply:
Actually, the ase_new with id 0 was previously never called. We discussed this on discord previously.
The client has to call the config with an id, and if its not already allocated that would then have happened in ase_get with a proper id already.

In regards to ASSERT: We would only return a NULL if we dont have enough memory, in which case we should respond with a BT_ASCS_RSP_NO_MEM. This is what happens now when ase_new returns NULL to ase_get, which forwards the NULL to ascs_config.

If the client calls with in invalid ASE, this is already handled in ascs_config, where we then respondBT_ASCS_RSP_INVALID_ASE

Reply:
Right, that is actually my point. Since we do the check for `id == 0` in `ascs_config`, where it isn't allowed, we don't need it here. 

Since `ase_get` may also be called internally, outside of `ascs_config`, then we should ensure that it is never 0. We could do that with this check too, but since it would only happen in case of a programming error on our side, an ASSERT might be more useful.

Reply:
I see the redundancy now.
However, `ase_get` is (no longer at least) called anywhere else. I will remove this check completely.